src/Compilers/Core/Portable/PEWriter/MetadataWriter.cs

```diff
@@ -65,25 +65,27 @@ internal abstract partial class MetadataWriter
         private readonly int _numTypeDefsEstimate;
         private readonly bool _deterministic;
 
-        // If true, it is allowed to have methods not have bodies (for emitting metadata-only assembly)
-        internal readonly bool allowMissingMethodBodies;
+        internal readonly bool MetadataOnly;
 
         // A map of method body before token translation to RVA. Used for deduplication of small bodies.
         private readonly Dictionary<ImmutableArray<byte>, int> _smallMethodBodies;
 
+        private static readonly ImmutableArray<byte> ThrowNullIL = ImmutableArray.Create((byte)ILOpCode.Ldnull, (byte)ILOpCode.Throw);
```

Comment:
>ThrowNullIL [](start = 53, length = 11)

Instead of hardcoding just the method IL, i'd suggest to hardocde the entire encoded method body byte sequence. It's gonna be just a single byte or two more and we can avoid some extra code in SerializeMethodBody.